src/telemetry/Manager.cc

```diff
@@ -38,6 +69,415 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
+	{
+	static auto string_vec_type = zeek::id::find_type<zeek::VectorType>("string_vec");
+	static auto double_vec_type = zeek::id::find_type<zeek::VectorType>("double_vec");
+	static auto count_vec_type = zeek::id::find_type<zeek::VectorType>("index_vec");
+	static auto histogram_metric_type = zeek::id::find_type<zeek::RecordType>(
+		"Telemetry::HistogramMetric");
+	static auto opts_idx = histogram_metric_type->FieldOffset("opts");
+	static auto labels_idx = histogram_metric_type->FieldOffset("labels");
+	static auto values_idx = histogram_metric_type->FieldOffset("values");
+	static auto count_values_idx = histogram_metric_type->FieldOffset("count_values");
+	static auto observations_idx = histogram_metric_type->FieldOffset("observations");
+	static auto sum_idx = histogram_metric_type->FieldOffset("sum");
+	static auto count_observations_idx = histogram_metric_type->FieldOffset("count_observations");
+	static auto count_sum_idx = histogram_metric_type->FieldOffset("count_sum");
+
+	zeek::RecordValPtr opts_record;
+
+	auto r = make_intrusive<zeek::RecordVal>(histogram_metric_type);
+
+	auto label_values_vec = make_intrusive<zeek::VectorVal>(string_vec_type);
+	for ( const auto& l : label_values )
+		label_values_vec->Append(make_intrusive<StringVal>(l));
+	r->Assign(labels_idx, label_values_vec);
+
+	auto fn = [&](const auto& histogram_data)
+	{
+		using val_t = std::decay_t<decltype(histogram_data.sum)>;
+		opts_record = telemetry_mgr->GetMetricOptsRecord<val_t>(MetricType::Histogram, family);
+		r->Assign(opts_idx, opts_record);
+
+		val_t observations = 0;
+		auto values_vec = make_intrusive<zeek::VectorVal>(double_vec_type);
+		auto count_values_vec = make_intrusive<zeek::VectorVal>(count_vec_type);
+
+		for ( const auto& b : histogram_data.buckets )
+			{
+			observations += b.count;
+			values_vec->Append(as_double_val(b.count));
+			if constexpr ( std::is_same_v<val_t, int64_t> )
+				count_values_vec->Append(val_mgr->Count(b.count));
+			}
+
+		r->Assign(values_idx, values_vec);
+		r->Assign(sum_idx, as_double_val(histogram_data.sum));
+		r->Assign(observations_idx, as_double_val(observations));
+
+		// Add extra fields just for int64_t based histograms with type count
+		if constexpr ( std::is_same_v<val_t, int64_t> )
+			{
+			r->Assign(count_values_idx, count_values_vec);
+			r->Assign(count_sum_idx, val_mgr->Count(histogram_data.sum));
+			r->Assign(count_observations_idx, val_mgr->Count(observations));
+			}
+	};
+
+	std::visit(fn, histogram);
+
+	// This is a bit annoying. The bounds / upper bounds of the individual
+	// buckets are not accessible through the family hdl. We set the
+	// opts$bounds field based on bounds collected from an individual
+	// metric instance.
+	//
+	// As the opts_record is cached by the manager, so below should only
+	// happen once.
+	//
+	// This is probably working around something that should be fixed
+	// in the broker::telemetry level.
```

Comment:
Seems worth creating a ticket for on the Broker-side to provide this, and then clean this up here. Shouldn't be too difficult I assume. (But ok to leave as is for now).